api/handlers/authentication.go

```diff
@@ -152,15 +152,9 @@ func (h Auth) CreateUser(c *gin.Context) {
 	err = redisDB.RDB.Set(context.Background(), userkey, userVerifyJSON, 24*time.Hour).Err()
 
 	//send OTP
-	err2 := sendOTP(user.Email, pin)
-	if err != nil || err2 != nil {
-		var temperr error
-		if err != nil {
-			temperr = err
-		} else {
-			temperr = err2
-		}
-		logger.WithError(temperr).Error("Error generating token")
+	go sendOTP(user.Email, pin)
+	if err != nil {
+		logger.WithError(err).Error("Error generating token")
 		c.JSON(http.StatusInternalServerError, models.Response{Error: "Error generating token"})
 		return
 	}
```

Comment:
This check handles the write to Redis on line 152. Please move it up and update the error message to reflect the error correctly